src/python_testing/matter_testing_infrastructure/nfc.py

```diff
@@ -0,0 +1,318 @@
+        return bytes(ndef_data)
+
+    def is_onboarding_data(self, ndef_bytes):
+        """
+        Checks if the NDEF message contains a URI record starting with 'MT:' (case-insensitive).
+
+        Args:
+            ndef_bytes (bytes): Raw NDEF message bytes.
+
+        Returns:
+            bool: True if a URI record starts with 'MT:' (case-insensitive), else False.
+        """
+        if not ndef_bytes:
+            logger.debug("No NDEF!")
+            return False
+        try:
+            for record in ndef.message_decoder(ndef_bytes):
+                if hasattr(record, "uri") and record.uri and record.uri.lower().startswith("mt:"):
+                    logger.debug(f"Found Matter URI: {record.uri}")
+                    return True
+        except Exception as e:
+            logger.error(f"NDEF parsing error: {e}")
+        return False
+
+    def write_t4t_ndef_uri(self, uri):
+        """
+        Writes an NDEF message with a single URI record to the NFC tag.
+
+        Args:
+            uri (str): The URI string to write.
+
+        Returns:
+            bool: True if write was successful, False otherwise.
+        """
+        with NFCConnectionManager(self) as connection:
+
+            # Select NDEF file (E104)
+            SELECT_NDEF = [0x00, 0xA4, 0x00, 0x0C, 0x02, 0xE1, 0x04]
+            _, sw1, sw2 = connection.transmit(SELECT_NDEF)
+            if (sw1, sw2) != (0x90, 0x00):
+                logger.debug(f"SELECT_NDEF failed: SW1=0x{sw1:02X}, SW2=0x{sw2:02X}")
+                return False
+
+            # Create NDEF message with a single URI record
+            record = ndef.UriRecord(uri)
+            ndef_message = b''.join(ndef.message_encoder([record]))
+            ndef_length = len(ndef_message)
+
+            # Write NDEF length (first 2 bytes)
+            WRITE_NLEN = [0x00, 0xD6, 0x00, 0x00, 0x02, (ndef_length >> 8) & 0xFF, ndef_length & 0xFF]
+            _, sw1, sw2 = connection.transmit(WRITE_NLEN)
+            if (sw1, sw2) != (0x90, 0x00):
+                logger.debug(f"WRITE_NLEN failed: SW1=0x{sw1:02X}, SW2=0x{sw2:02X}")
+                return False
+
+            # Write the NDEF message itself
+            # The maximum length per APDU is usually 255 bytes; split if needed
+            offset = 0
+            while offset < ndef_length:
+                chunk = ndef_message[offset:offset+0xFF]
+                chunk_len = len(chunk)
+                WRITE_NDEF = [0x00, 0xD6, (offset + 2) >> 8, (offset + 2) & 0xFF, chunk_len] + list(chunk)
+                _, sw1, sw2 = connection.transmit(WRITE_NDEF)
+                if (sw1, sw2) != (0x90, 0x00):
+                    logger.debug(f"WRITE_NDEF failed at offset {offset}: SW1=0x{sw1:02X}, SW2=0x{sw2:02X}")
+                    return False
+                offset += chunk_len
+
+        logger.info(f"Successfully wrote URI '{uri}' to NFC tag.")
+        return True
+
+
+class NFCConnectionManager:
```

Comment:
```suggestion
class NFCConnection:
```

Seems to maybe create more readable code like:

```python
with NFCConnection(reader) as connection:
   # ...
```

I am unsure if manager adds to the usage explanation. This is a `createConnection + disconnect`